libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -393,6 +398,15 @@ class MapboxNavigation(
         roadObjectMatcher = RoadObjectMatcher(navigator)
     }
 
+    /**
+     * Provides the current navigation session state.
+     *
+     * @see NavigationSessionState
+     * @see NavigationSessionStateObserver
+     * @see [registerNavigationSessionStateObserver]
+     */
+    val navigationSessionState = navigationSession.state
```

Comment:
This is a variable because of the underlying implementation of `navigationSession.state`, not an immutable value as the declaration would suggest. I'd suggest making this a function:
```suggestion
    fun getNavigationSessionState(): NavigationSessionState = navigationSession.state
```